lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -1621,16 +1621,18 @@ namespace Js
 
     BOOL JavascriptProxy::GetDiagTypeString(StringBuilder<ArenaAllocator>* stringBuilder, ScriptContext* requestContext)
```

Comment:
>JavascriptProxy::GetDiagTypeString [](start = 9, length = 34)

how did you identify which of the methods needed to use the new helper? how do we know we are not missing other cases and how do we prevent new cases being introduced?

Reply:
There was a regression caused by my previous change. I also noticed another place using exactly the same pattern.
Simply deferring to the target is actually not normally done here.
We typically do analysis on the proxy itself or dig out the target and do analysis on it.

We also may call operators on the target. That _is_ a recursive pattern, but I guess it is supposed to be recursive. Such pattern is very common here.


---
In reply to: [205296417](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205296417) [](ancestors = 205296417)

Reply:
The recursion via operators is fine because we do `PROBE_STACK` when implementing those for proxies.
@akroshg would know for sure.

---
In reply to: [205636755](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205636755) [](ancestors = 205636755,205296417)